libusb.go

```diff
@@ -224,14 +224,21 @@ func (libusbImpl) setDebug(c *libusbContext, lvl int) {
 }
 
 func (libusbImpl) getDeviceDesc(d *libusbDevice) (*DeviceDesc, error) {
-	var desc C.struct_libusb_device_descriptor
+	var desc, pathData, path = C.struct_libusb_device_descriptor{}, [8]uint8{}, []uint8{}
 	if err := fromErrNo(C.libusb_get_device_descriptor((*C.libusb_device)(d), &desc)); err != nil {
 		return nil, err
 	}
+	pathLen := int(C.libusb_get_port_numbers((*C.libusb_device)(d), (*C.uint8_t)(&pathData[0]), 8))
+	if pathLen > 0 {
+		path = pathData[:pathLen-1]
+	}
```

Comment:
to put ints in the Path (rather than `uint8`s), you'll need to make a copy instead.

Reply:
Copy may be neater than my for loop but I couldn't see a way to make that work?

Reply:
sorry, I didn't mean to use `copy()`, just create a copy of data with converted types.

Reply:
generally, what you did in the for loop is correct, that's what I meant.